test/SILOptimizer/globalopt_resilience.swift

```diff
@@ -21,15 +21,38 @@ public struct ResilientStruct {
 }
 
 public func cannotConvertToValueUse() {
-  // We can't optimize this because the method takes the resilient value as
-  // @in_guaranteed
+  // Previously we could not optimize this because the method takes the
+  // resilient value as @in_guaranteed. But now SILGen produces better
+  // code for us up-front, so now its optimized anyway.
+  ResilientStruct.staticVal.method()
+}
+
+@inlinable public func cannotConvertToValueUseAlt() {
+  // SILGen still produces address-only code in the inlinable case.
+  // We also optimize this correctly, but with a slightly different
+  // pattern.
   ResilientStruct.staticVal.method()
 }
 
 // CHECK-LABEL: sil @$s4test23cannotConvertToValueUseyyF : $@convention(thin) () -> ()
-// CHECK: [[ADDR:%.*]] = global_addr @$s4test15ResilientStructV9staticValACvpZ : $*ResilientStruct
+// CHECK: [[INT:%.*]] = integer_literal $Builtin.Int{{32|64}}, 27
+// CHECK: [[S1:%.*]] = struct $Int ([[INT]] : $Builtin.Int{{32|64}})
+// CHECK: [[S2:%.*]] = struct $ResilientStruct ([[S1]] : $Int)
+// CHECK: [[TMP:%.*]] = alloc_stack $ResilientStruct
+// CHECK: store [[S2]] to [[TMP]] : $*ResilientStruct
+// CHECK: [[METHOD:%.*]] = function_ref @$s4test15ResilientStructV6methodyyF : $@convention(method) (@in_guaranteed ResilientStruct) -> ()
+// CHECK: apply [[METHOD]]([[TMP]]) : $@convention(method) (@in_guaranteed ResilientStruct) -> ()
+// CHECK: [[RESULT:%.*]] = tuple ()
+// CHECK: return [[RESULT]] : $()
+
+// CHECK-LABEL: sil @$s4test26cannotConvertToValueUseAltyyF : $@convention(thin) () -> ()
+// CHECK: [[TMP:%.*]] = alloc_stack $ResilientStruct
```

Comment:
I would like at least a FIXME here.

FIXME: The optimizer should easily eliminate this copy. The problem is that copy forwarding assumes that address-only types are always explicitly destroyed.